### PR TITLE
Issues with the list in Spike view [SDESK-5923]

### PR DESCRIPTION
**Pull request by GyanP**
Resolve constant loading.

### CODE REVIEW

File: scripts/core/itemList/LazyLoader.tsx [hunk 1]
```diff
@@ -63,7 +63,7 @@ export class LazyLoader<T> extends React.Component<IProps<T>, IState<T>> {
     private reloadAllItems() {
         const MAX_PAGE_SIZE = 200; // back-end limit
 
-        const loadedItemsCount = this.state.items.size;
+        const loadedItemsCount = this.state.items.size > 0 ? this.state.items.size : 1;
```
Comment (petrjasek) [inline scripts/core/itemList/LazyLoader.tsx:66]: can you pls add some comment to this? not sure why you do it

**Comment by GyanP** [comment]
We need a default value of 1 because if we get 0 then it will generate an Array with 0 values [here](https://github.com/superdesk/superdesk-client-core/blob/develop/scripts/core/itemList/LazyLoader.tsx#L68)  which we don't want, else it will break down [here](https://github.com/superdesk/superdesk-client-core/blob/develop/scripts/core/itemList/LazyLoader.tsx#L84) with an error that `cannot reduce an empty array.`

File: scripts/core/itemList/LazyLoader.tsx (hunk 1 above)
Comment (GyanP) [reply scripts/core/itemList/LazyLoader.tsx:66]: Done.

**Comment by petrjasek** [comment]
imo if the size is 0 it should just return, there is no point to do any reloading:
```ts
if (loadedItemsCount === 0) {
    return;
}
```

**Comment by GyanP** [comment]
> imo if the size is 0 it should just return, there is no point to do any reloading:

We need to reload it because of the new item. The size doesn't contain the new items, it's just the size of loadedItemsCount (that already present) and for the first time it would be 0 because there are no items present, so If the new item is being sent so it should reload.

**Comment by petrjasek** [comment]
I see, can you add that info to the code?

**Comment by GyanP** [comment]
> I see, can you add that info to the code?

Done.

**Comment by petrjasek** [comment]
shouldn't we add +1 all the time then to get new items? and not just when it was empty? because it doesn't check for the total number of items so if it would go from MAX_PAGE_SIZE count to MAX_PAGE_SIZE + 1 that extra item would be me missing right?

**Review by petrjasek — APPROVED** [review]

**Pull request merged**